ivadomed/random.py

```diff
@@ -0,0 +1,13 @@
+import torch
+import random
+import numpy as np
+
+def set_seed(seed: int = 6):
+    """
+    This function controls sources of randomness to aid reproducibility.
+    Args:
+        seed: int to initialize random number generator (RNG)
+    """
+    random.seed(seed)
+    np.random.seed(seed)
+    torch.manual_seed(seed)
```

Comment:
Just adding a reference here - https://pytorch-lightning.readthedocs.io/en/stable/_modules/pytorch_lightning/utilities/seed.html#seed_everything to set the seed across the GPUs also.
I think that also should be added here.

Reply:
This is also a good reference about setting global seed - https://github.com/chandar-lab/RLHive/blob/c5b1b77a3daf87aeb877e10c4f65bcd7e2bc1ff8/hive/utils/utils.py#L34

Reply:
Thank you @naga-karthik. 
With the current implementation of `random.py`, I am able to reproducible the results of an experiment on CPU. I ran this [tutorial](https://ivadomed.org/tutorials/one_class_segmentation_2d_unet.html) with this [config.json](https://github.com/ivadomed/ivadomed/blob/master/ivadomed/config/config.json) for 8 runs using the same seed, 5 epochs each and I obtained the same results.

Also, the experiment was done with the seed set inside the [main.py](https://github.com/etetteh/ivadomed/blob/a7125d5e5f3c946eefc2f9aec8b92e505d6c6244/ivadomed/main.py#L351)  only, without those in [loader/utils.py](https://github.com/etetteh/ivadomed/blob/a7125d5e5f3c946eefc2f9aec8b92e505d6c6244/ivadomed/loader/utils.py#L145), and [training.py](https://github.com/etetteh/ivadomed/blob/a7125d5e5f3c946eefc2f9aec8b92e505d6c6244/ivadomed/training.py#L58).

However, I have not been able to obtain reproducible results on a GPU (V100), even upon adding the GPU specific features referenced above by @naga-karthik

Reply:
I reviewed a few more widely used libraries which could serve as references:

- MONAI - https://github.com/Project-MONAI/MONAI/blob/a676e3876903e5799eb3d24245872ea4f3a5b152/monai/utils/misc.py#L224
-  MMSegmentation - https://github.com/open-mmlab/mmsegmentation/blob/master/mmseg/apis/train.py#L19

Reply:
Thanks @kanishk16. I will run experiments with those suggestions, to see if they fix the issue.

Previously, I tested the [torch.backends.cudnn.deterministic = True](https://github.com/Project-MONAI/MONAI/blob/a676e3876903e5799eb3d24245872ea4f3a5b152/monai/utils/misc.py#L275), but it raises an error for some objects in ivadomed, because not all objects have a deterministic capability. Will dive deeper.